### ISSUE TITLE
When will there be v0.8?

### ISSUE THREAD

**Issue by SamuelBorn** [libratbag/piper#832]
I need v0.8 of piper for my mouse (Logitech MX Master 3s).

I dont want to go thorugh the hassle of compiling it myself. 

As the last release is now 7 months old I was wondering when we can expect the next release?

**Comment by staticssleever668** [comment]
> I need v0.8 of piper for my mouse (Logitech MX Master 3s).

You actually only need a device file for that! :)

Crate a file named `logitech-MX-Master-3S.device` under `/usr/share/libratbag` or /usr/local/share/libratbag` (depending on where other device files are located). The file path will look like `/usr/share/libratbag/logitech-MX-Master-3S.device`.
Put this text into this new file:

```dosini
[Device]
DeviceMatch=bluetooth:046d:b034
Driver=hidpp20
Name=Logitech MX Master 3S
```

And restart ratbagd service, for which I suggest simply running `systemctl restart ratbagd` in the terminal.

(related PR https://github.com/libratbag/libratbag/pull/1382).

---

As for the original question, there will be a new release once we finalize the next version of libratbag API and test it a little (tracking issue https://github.com/libratbag/libratbag/issues/1388).
Currently only I more or less actively maintain libratbag, so this depends completely on me. I'd say this would take me 2-3 months of doing things in free time, but I have some stuff going on in my life at the moment, so it will take a bit more.

**Comment by SamuelBorn** [comment]
> in free time
You are awesome! Thank you for your contributions to the open source community.

---

I made it work, with the changes you said. Now I can remap keys, the way I want.

The only bummer is. It only worked via bluetooth and not my 2.4Ghz adapter.
I also looked [here](https://github.com/libratbag/libratbag/tree/master/data/devices) to maybe find a 2.4Ghz Logitech MX Master 3S device file, but didnt find one. 
So I guess 2.4Ghz is not supported?

**Comment by staticssleever668** [comment]
> The only bummer is. It only worked via bluetooth and not my 2.4Ghz adapter.

You can try making another device file with following contents:

```
[Device]
Name=Logitech Bolt Receiver
DeviceMatch=usb:046d:c548
Driver=hidpp20
```

Adding device files for receivers is said to bring some problems, but I can only think of it potentially making the device appearing as two separate devices in Piper.
We generally rely on Linux reexposing receivers as proper devices, but if I understand right Linux doesn't know about some of the newer receivers (yet).

**Comment by SamuelBorn** [comment]
This unfortunetly did not help.  After creating the file and restarting ratbagd I get the message "Cannot find any devices".

But  it isnt that important. Im relatively fine with bluetooth working.

**Comment by rickoman** [comment]
> You actually only need a device file for that!

Not exactly... Piper v0.7 (latest) will not talk to ratbagd v0.17 (latest) because the protocol has changed v1 to v2 in ratbagd.  And you need ratbagd v0.17 because it supports the newer Bolt Receiver.

I have not had much luck with Bluetooth and my MX 3s.

Any chance for Piper v0.7.x snapshot release with protocol v2 support so at least the latest of each project can talk to each other?

**Comment by SamuelBorn** [comment]
> I have not had much luck with Bluetooth and my MX 3s.

I am using the MX 3s and can say that the device file is definitely sufficient!
Try it with Bluetooth.
I also couldn't get the 2.4Ghz to work, though.

**Comment by staticssleever668** [comment]
> Piper v0.7 (latest) will not talk to ratbagd v0.17 (latest)

That's not really right, they are both at API level 1. Perhaps you are running a git version of libratbag?

> And you need ratbagd v0.17 because it supports the newer Bolt Receiver.

I don't think we support receivers at all. We expect the Linux kernel to re-export devices connected to receivers as separate USB devices, and only them do we support.

---

Either way, I suggest you reinstall libratbag and Piper from distro's repository and try the aforementioned way again (https://github.com/libratbag/piper/issues/832#issuecomment-1407412783).
It probably won't really make any difference if you compile them from sources, as I don't remember any changes to the Logitech driver after the latest release.

**Comment by toupeira** [comment]
Friendly ping, any news on this?

I'm using the Debian unstable packages, which has a recent `ratbagd` 0.17 with the device file for my Logitech G502 X Lightspeed, but `piper` 0.7 with a missing SVG. Everything seems to be working fine though, just a small hurdle figuring out which body part of the cute mouse corresponds to which button :smile:
